plugins/generic/pln/classes/Deposit.inc.php

```diff
@@ -97,6 +97,95 @@ function setStatus($status) {
 	}
 
 	/**
+	 * Return a string representation of the local status.
+	 * 
+	 * @param $translatable boolean
+	 * @return string
+	 */
+	function getLocalStatus($translatable = true) {
+		$status = '';
+		if($this->getNewStatus()) {
+			$status = 'new';
+		}
+		if($this->getPackagedStatus()) {
+			$status = 'packaged';
+		}
+		if($this->getTransferredStatus()) {
+			$status = 'transferred';
+		}
+		
+		if($translatable) {
+			return PLN_PLUGIN_LOCALE_PREFIX . $status;
+		}
+		return $status;
+	}
+	
+	/**
+	 * Return a string representation of the processing status.
+	 * 
+	 * @param $translatable boolean
+	 * @return string
+	 */
+	function getProcessingStatus($translatable = true) {
+		$status = 'unknown';
+		if($this->getReceivedStatus()) {
+			$status = 'received';
+		}
+		if($this->getValidatedStatus()) {
+			$status = 'validated';
+		}
+		if($this->getSentStatus()) {
+			$status = 'sent';
+		}
+		if($translatable) {
+			return PLN_PLUGIN_LOCALE_PREFIX . $status;
```

Comment:
I'm not a big fan of building locale keys programmatically, because it makes it hard to find where a key is used in the code (e.g. by grepping). Suggest rewriting these functions into a switch, e.g.:

````
switch (true) {
    case $this->getSentStatus(): return 'plugins.generic.pln.status.sent';
    case $this->getValidatedStatus(): return 'plugins.generic.pln.status.validated';
    ...etc...
}
return 'plugins.generic.pln.status.unknown';
````
Nice and compact, but doesn't render locale strings un-greppable.